crates/kumod/src/lua_deliver.rs

```diff
@@ -58,6 +58,10 @@ impl QueueDispatcher for LuaQueueDispatcher {
     }
 
     async fn attempt_connection(&mut self, dispatcher: &mut Dispatcher) -> anyhow::Result<()> {
+        if self.connection.is_some() {
+            dispatcher.delivered_this_connection = 0;
```

Comment:
This line would appear to disable max_deliveries_per_connection; is that intentional?

Reply:
yeah, I thought it would be better to just keep using the same connection and let the maintainer_task close it if there's no messages anymore.

Reply:
I'd like to keep some kind of limit here to defend against pathological situations if the system is ever under extreme stress.  You should be able to configure this differently via the egress path configuration if you'd like to use more than the default of 1024 per connection.
Can we remove this line from the PR?

Reply:
for sure, I'll push an update to remove the line